File: pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> pascal_triangle;
        for(int i=1;i<=rowIndex+1;i++)
        {
            vector<int> temp(i);
            temp[0] = 1;
            temp[i-1] = 1;
            pascal_triangle.push_back(temp);
        }
        for(int i=2;i<=rowIndex;i++)
            for(int j=1;j<i;j++)
                pascal_triangle[i][j] = pascal_triangle[i-1][j-1] + pascal_triangle[i-1][j];
        return pascal_triangle[rowIndex];
    }
};

File: pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        Integer[] curr = new Integer[rowIndex+1];
        Arrays.fill(curr, 0);
        curr[0] = 1;
        for(int n=0;n<=rowIndex;n++)
        {
            for(int j=n;j>0;j--)
                curr[j] += curr[j-1];
            
        }
        return Arrays.asList(curr);
    }
}
